compiler/rustc_index/src/vec.rs

```diff
@@ -156,6 +176,8 @@ macro_rules! newtype_index {
             }
 
             /// Extracts the value of this index as a `usize`.
+            ///
+            /// This casts a `u32` to a `usize`.
```

Comment:
nit: Why did you add this comment here but not also for `index`? I think it should either be in both places or neither.